app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
We still need to be able to record the chains, because the input of the dependent task doesn't have to match exactly the output of it's dependency. For examples, let's have an action, that produces a host, and other action, that has just an ip on it's input:

```
  host_creation = plan_action(CreateHost, host_attrs)
  plan_action(DhcpReserve, host_creation.output[:mac])
```
I do think the chains make the concept more flexible and allows to combine the actions easier. i don't see a big difference in `output` vs. `result` in planning, we could introduce the `result` but I'm not sure it would help anything.